lib/src/https.rs

```diff
@@ -1886,12 +1890,113 @@ impl Proxy {
         }
     }
 
-    pub fn remove_listener(&mut self, address: StdSocketAddr) -> bool {
+    pub fn remove_listener(
+        &mut self,
+        remove: RemoveListener,
+    ) -> anyhow::Result<Option<ProxyResponseContent>> {
         let len = self.listeners.len();
 
         self.listeners
-            .retain(|_, listener| listener.borrow().address != address);
-        self.listeners.len() < len
+            .retain(|_, listener| listener.borrow().address != remove.address);
+
+        if !self.listeners.len() < len {
+            bail!(
+                "no HTTPS listener to remove at address {:?}",
+                remove.address
+            )
+        }
+        Ok(None)
+    }
+
+    pub fn soft_stop(&mut self) {
```

Comment:
It could be interesting to return a `Result` here

Reply:
sure why not